File: server/src/main/java/ru/fullrest/mfr/server/model/repository/PropertyRepository.java
package ru.fullrest.mfr.server.model.repository;

import org.springframework.data.repository.CrudRepository;
import ru.fullrest.mfr.server.model.entity.Property;
import ru.fullrest.mfr.server.model.entity.PropertyType;

import java.util.Optional;

public interface PropertyRepository extends CrudRepository<Property, Integer> {

    Optional<Property> findByType(PropertyType type);
}

File: server/src/main/kotlin/com/lezenford/mfr/server/model/repository/PropertyRepository.kt
package com.lezenford.mfr.server.model.repository

import com.lezenford.mfr.server.model.entity.Property
import org.springframework.data.jpa.repository.JpaRepository

interface PropertyRepository : JpaRepository<Property, Int> {
    fun findByType(type: Property.Type): Property?
}